https

```diff
@@ -51,15 +55,30 @@ IP6=$( ip -6 addr show scope global | grep -o '\(inet6 \)[^/]\+' -m 1 | cut -c 7
 
 [ -z $IP4 ] && echo "ipv4: not found" || echo "ipv4: ${IP4}"
 [ -z $IP6 ] && echo "ipv6: not found" || echo "ipv6: ${IP6}"
-printf "\n"
-
-cf_api="curl -s -H X-Auth-Email:${CF_Email} -H X-Auth-Key:${CF_Key} ${CF_Endpoint}"
+echo ""
+
+# $1: path (e.g. zones/<id>/dns_records)
+# $@: args appended to `curl`
+function cf_api () {
+    local path="$1"
+    shift 1
+
+    if [ -n "${CF_Email}" -a -n "${CF_Key}" ] ; then
+        curl -s -H "X-Auth-Email: ${CF_Email}" -H "X-Auth-Key: ${CF_Key}" "${CF_Endpoint}${path}" "$@"
+    elif [ -n "${CF_Account_ID}" -a -n "${CF_Token}" ] ; then
+        curl -s -H "Authorization: Bearer ${CF_Token}" "${CF_Endpoint}${path}" "$@"
+    else
+        echo "assertion: not reached."
+	echo "either CF_Email + CF_Key or CF_Account_ID + CF_Token needs to be provided"
+        exit 2
```

Comment:
`return 2`?
```
> bash -c 'f() { exit 1; }; f; echo 1'
> bash -c 'f() { return 1; }; f; echo 1'
1
```

Reply:
`exit` is fine.

else branch shouldn't be reachable (checked at line 11)